### PR TITLE
Adding ManimCommunity colors

### PR DESCRIPTION
**Pull request by kolibril13**
## Motivation
Added an extra file for the community colours.
## Overview / Explanation for Changes

With the current master, one can only use colours like this:
```py
from manim import *

class ManimCELogoNO(Scene):
    def construct(self):
        self.camera.background_color = WHITE
        ds_m = MathTex(r"\mathbb{M}", fill_color=BLACK).scale(7)
        ds_m.shift(2.25 * LEFT + 1.5 * UP)
        circle = Circle(color=GREEN, fill_opacity=1).shift(LEFT)
        square = Square(color=BLUE, fill_opacity=1).shift(UP)
        triangle = Triangle(color=RED, fill_opacity=1).shift(RIGHT)
        logo = VGroup(triangle, square, circle, ds_m)  # order matters
        logo.move_to(ORIGIN)
        self.add(logo)
   ```
ManimCELogoNO

And with this pr, it is also possible to quickly change the colourmap like this:
```python
from manim import *
from manim.communitycolors import *

class ManimCELogo(Scene):
    def construct(self):
        self.camera.background_color = WHITE
        ds_m = MathTex(r"\mathbb{M}", fill_color=BLACK).scale(7)
        ds_m.shift(2.25 * LEFT + 1.5 * UP)
        circle = Circle(color=GREEN, fill_opacity=1).shift(LEFT)
        square = Square(color=BLUE, fill_opacity=1).shift(UP)
        triangle = Triangle(color=RED, fill_opacity=1).shift(RIGHT)
        logo = VGroup(triangle, square, circle, ds_m)  # order matters
        logo.move_to(ORIGIN)
        self.add(logo)
```
image

I really like this approach because it enables one to quickly change colours, without the necessity to change code.
However, an issue would be that e.g. an extra ` self.add((Dot()))` would add a white Dot, as white is the default dot colour. But this would still be the normal white, not the community white.
Also, there are still all the other colours, e.g. GREEN_A, GREEN_B etc. which are not touched by the line `from manim.communitycolors import *`  and would lead to some inconsistency.
Another option would be to just add the community colours to the other colours, e.g. as GREEN_MC, RED_MC etc.
However, I am not a big fan of that, as there are already a lot of colours in the normal namespace.

Something that also comes into my mind is that we should maybe only have e.g. 10 base colours (RED, GREEN, YELLOW, etc.) that are overrideable and all other colours can be added with an extra import (RED_A, RED_B, GREEN_A, GREEN_B etc..)

Another point I want to note is that the colours are right now not completely consistent.
E.g. for the normal colors, there are A,B,C,D but then then there is     "DARK_GREY", "DARK_GRAY","DARKER_GREY", "DARKER_GRAY", and only one ORANGE and two pinks.

## Oneline Summary of Changes

```
- Adding ManimCommunity colors  (:pr:`PR NUMBER HERE`)
```

## Acknowledgements
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)

## Reviewer Checklist
- [ ] Newly added functions/classes are either private or have a docstring
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings
- [ ] The oneline summary has been included [in the wiki](https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release)

### CODE REVIEW

File: manim/communitycolors.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+"""The colormap of manim community"""
+
+WHITE = "#ece6e2"
+GREEN = "#87c2a5"
+BLUE = "#525893"
+RED = "#e07a5f"
+BLACK = "#343434"
```
Comment (naveen521kk) [inline manim/communitycolors.py:7]: Move it into the place where colours are defined. There can't be a file just for this. 
Move it to the Color Enum we have and name it as `LOGO_WHITE`,`LOGO_RED` etc.

**Review by naveen521kk — CHANGES_REQUESTED** [review]

File: manim/communitycolors.py (hunk 1 above)
Comment (kolibril13) [reply manim/communitycolors.py:7]: I think the community colours should be importable like
`from manim.communitycolors import *` and not in the default namespace, that's why I put it there.

**Comment by kolibril13** [comment]
One further thought about the colours:
Maybe there can be a plugin that allows using matplotlib colours, here is a draft of how that could be used:
```python
class ABC(Scene):
    def construct(self):
        self.camera.background_color = WHITE
        colors = ["mediumspringgreen", "skyblue", "dodgerblue", "navy"]
        mediumspringgreen, skyblue, dodgerblue, navy = (
            mpl.colors.to_hex(col) for col in colors
        )
        d = Dot(color=mediumspringgreen)
        c = Circle(color=skyblue)
        s = Square(color=dodgerblue)
        t = Triangle(color=navy)
        d.next_to(c, RIGHT)
        s.next_to(c, LEFT)
        t.next_to(c, DOWN)
        self.add(d, c, s, t)
```
ABC

**Comment by leotrs** [comment]
I like the idea of adding the CE colors, but honestly I would be in favor of engineering a solution that's a bit more fully-featured in order to cover some of the cases that you mention (instead of just adding a file that overrides some colors).

What we really need is a system for easily choosing and changing color palettes.

Related to #310

**Comment by kolibril13** [comment]
> What we really need is a system for easily choosing and changing color palettes.

I totally agree with this!
As a quick and temporary solution, I suggest that the community colors are not overwriting the default colors, but have the names LOGO_WHITE, LOGO_RED etc. .
Additionally, I like to stay with the idea that they are not going to the enum, but have to be imported separately from the manim/colors file.

File: manim/communitycolors.py (hunk 1 above)
Comment (kolibril13) [inline manim/communitycolors.py:7]:
```suggestion
LOGO_WHITE = "#ece6e2"
LOGO_GREEN = "#87c2a5"
LOGO_BLUE = "#525893"
LOGO_RED = "#e07a5f"
LOGO_BLACK = "#343434"
```

File: manim/communitycolors.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+"""The colormap of manim community"""
+
+<span class="x x-first x-last">LOGO_WHITE</span> = "#ece6e2"
+<span class="x x-first x-last">LOGO_GREEN</span> = "#87c2a5"
+<span class="x x-first x-last">LOGO_BLUE</span> = "#525893"
+<span class="x x-first x-last">LOGO_RED</span> = "#e07a5f"
+<span class="x x-first x-last">LOGO_BLACK</span> = "#343434"
```
Comment (kolibril13) [inline manim/communitycolors.py:7]:
```suggestion
LOGO_WHITE = "#ece6e2"
LOGO_GREEN = "#87c2a5"
LOGO_BLUE = "#525893"
LOGO_RED = "#e07a5f"
LOGO_BLACK =  "#343434"
```

**Comment by leotrs** [comment]
What is the benefit of having them in a separate file if they already have different names (that won't overwrite anything on the Enum file)?

**Comment by kolibril13** [comment]
They belong to a different colormap than the default colors.
Therefore, my idea was to not have them imported by default.

**Comment by leotrs** [comment]
How about a different Enum on the same file? :thinking: Or something that would allow us to do `from manim.colors import community_colors` or something similar. (I think for this we *would* need a different file...)

**Comment by kolibril13** [comment]
> How about a different Enum on the same file?  Or something that would allow us to do `from manim.colors import community_colors` or something similar. (I think for this we _would_ need a different file...)

`from manim.communitycolors import *` is already possible with the current approach.
You are right, one could make an extra Enum in the file where the other colors are.
However, then we would still need one extra file to import this enum to make this `from manim.colors import community_colors` work.

> What we really need is a system for easily choosing and changing colour palettes.


Yep. As this is a longterm goal, and as we do not have that yet, are you fine with the current solution as a temporary solution, at least to have the community colours somewhere?

**Comment by leotrs** [comment]
As long as we come back to it and actually implement that, then I'm all for this being merged

**Comment by kolibril13** [comment]
> As long as we come back to it and actually implement that, then I'm all for this being merged

Nice!
@naveen521kk : are you also fine with this temporary solution?

File: manim/communitycolors.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+"""The colormap of manim community"""
+
+LOGO_WHITE = "#ece6e2"
+LOGO_GREEN = "#87c2a5"
+LOGO_BLUE = "#525893"
+LOGO_RED = "#e07a5f"
+LOGO_BLACK =  "#343434"
```
Comment (kolibril13) [inline manim/communitycolors.py:7]:
```suggestion
LOGO_BLACK = "#343434"
```

**Comment by kolibril13** [comment]
Can this then be approved, @naveen521kk ?

**Review by naveen521kk — APPROVED** [review]

**Comment by kolibril13** [comment]
Why are tests for macos-3.8 failing?
```
==> tesseract
This formula contains only the "eng", "osd", and "snum" language data files.
If you need any other supported languages, run `brew install tesseract-lang`.
==> pyenv-virtualenv
To enable auto-activation add to your profile:
  if which pyenv-virtualenv-init > /dev/null; then eval "$(pyenv virtualenv-init -)"; fi
Error: Calling brew cask install is disabled! Use brew install [--cask] instead.
Error: Process completed with exit code 1.
```
Probably an github issue?
I ll run the tests again tomorrow.

**Pull request merged**